erpnext/agriculture/doctype/harvest/harvest.py

```diff
@@ -3,8 +3,53 @@
 # For license information, please see license.txt
 
 from __future__ import unicode_literals
-# import frappe
+import frappe
+import json
+from frappe import _
 from frappe.model.document import Document
 
 class Harvest(Document):
 	pass
+
+@frappe.whitelist()
+def create_stock_entry(harvest):
+	harvest = frappe.get_doc("Harvest", harvest)
+	if stock_entry_exists(harvest.get('name')):
+		stock_entry_name = frappe.db.get_value("Stock Entry", {"harvest": harvest.get('name')}, fieldname = ['name'])
+		return frappe.msgprint(_('Stock Entry {0} has been already created against this harvest.').format(frappe.utils.get_link_to_form("Stock Entry", stock_entry_name)))
+	stock_entry = frappe.new_doc('Stock Entry')
+	stock_entry.harvest = harvest.get('name')
+	stock_entry.purpose = harvest.get('purpose')
+	stock_entry.stock_entry_type = 'Harvest'
+
+	if harvest.get('strain'):
+		stock_entry = update_stock_entry_based_on_strain(harvest, stock_entry)
+
+	if stock_entry.get("items"):
```

Comment:
If there's no stock entry items, what are we making it for? :sweat_smile:

Reply:
had discussion with @neilLasrado  Regarding this. we need to make strain produced item and byproduct mandatory then or we need to add validation on produce_items and byproduct items

Reply:
Only Produced Items should be mandatory.